routing/control_tower.go

```diff
@@ -174,13 +181,12 @@ func (p *controlTower) Fail(paymentHash lntypes.Hash,
 		return err
 	}
 
-	// Notify subscribers of fail event.
-	p.notifyFinalEvent(
-		paymentHash, createFailedResult(
-			payment.HTLCs, reason,
-		),
-	)
-
+	// If recording the failure reason took the payment to a final state,
+	// we should notify all subscribers of the final payment status. This
+	// can happen if the payment had no more attempts in flight at this
+	// point, and recording this payment failure took the payment to a
+	// final failure state.
+	p.notifyFinalEvent(payment)
```

Comment:
If we don't notify here, because there are still htlcs in-flight, is this `Fail` method guarantueed to be called again? (does every failed htlc eventually trigger a call to this method?)

Reply:
No. That's why we also call it in `FailAttemp` above

Reply:
Aha, clear.